uni/lib/view/library/library.dart

```diff
@@ -0,0 +1,97 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_redux/flutter_redux.dart';
+import 'package:percent_indicator/linear_percent_indicator.dart';
+import 'package:tuple/tuple.dart';
+import 'package:uni/model/app_state.dart';
+import 'package:uni/model/entities/library_occupation.dart';
+import 'package:uni/view/common_widgets/page_title.dart';
+import 'package:uni/view/common_widgets/pages_layouts/general/general.dart';
+import 'package:uni/view/common_widgets/request_dependent_widget_builder.dart';
+import 'package:uni/view/library/widgets/library_occupation_card.dart';
+
+class LibraryPageView extends StatefulWidget {
+  const LibraryPageView({Key? key}) : super(key: key);
+
+  @override
+  State<StatefulWidget> createState() => LibraryPageViewState();
+}
+
+class LibraryPageViewState extends GeneralPageViewState<LibraryPageView> {
+  @override
+  Widget getBody(BuildContext context) {
+    return StoreConnector<AppState, Tuple2<LibraryOccupation, RequestStatus>>(
+        converter: (store) {
+      final LibraryOccupation occupation =
+          store.state.content['libraryOccupation'];
+      return Tuple2(occupation, store.state.content['libraryOccupationStatus']);
+    }, builder: (context, occupationInfo) {
+      return RequestDependentWidgetBuilder(
+          context: context,
+          status: occupationInfo.item2,
+          contentGenerator: generateOccupationPage,
+          content: occupationInfo.item1,
+          contentChecker: occupationInfo.item1.capacity != 0,
+          onNullContent: Center(
+              child: Text('Não existem dados para apresentar',
+                  style: Theme.of(context).textTheme.headline4,
+                  textAlign: TextAlign.center)));
+    });
+  }
+
+  Widget generateOccupationPage(occupation, context) {
+    return ListView(
+        scrollDirection: Axis.vertical,
+        shrinkWrap: true,
+        children: <Widget>[
+          const PageTitle(name: 'Biblioteca'),
+          LibraryOccupationCard(),
+          const PageTitle(name: 'Pisos'),
+          createFloorRow(occupation.getFloor(1), occupation.getFloor(2)),
+          createFloorRow(occupation.getFloor(3), occupation.getFloor(4)),
+          createFloorRow(occupation.getFloor(5), occupation.getFloor(6))
+        ]);
```

Comment:
Create a function to getFloorRows that uses a loop to create this array.
Makes this more scalable. (The number of floors won't change, however it's good practice to do this way)